composer.json

```diff
@@ -13,14 +13,12 @@
     "require": {
```

Comment:
`paragonie/random_compat` is a polyfill for `random_bytes()` and `random_int()` - both these are native functions in PHP 7+. This package is _only_ required for PHP 5 support. As we don't support PHP 5, there's no need for this library at all.

We explicitly add it to the `replace` to stop the dependencies that do support PHP 5 from installing an empty package - `paragonie/random_compat` installs an empty package on PHP 7.

Reply:
Ah OK - at least now I get the thinking! Problem is this is going to conflict with anything that specifies the library (empty package version or not). This is what we've just been seeing when bringing the new govuk account stuff into VOL.

Reply:
@jerotire might have the specific error noted down somewhere. But it amounted to if something else wants to install paragonie/random_compat it can't do that, because composer believes dvsa/contracts is a replacement. Differing versions of the same package can't be installed twice as it were. This doesn't have to be a problem for our code, but it might be for someone who is using our code. The confusion we've had today could be enjoyed by others, and it's not worth that so we can avoid installing an empty package which causes no harm.

Either way we're labelling our code as being a replacement for a package that it isn't a replacement for. If we were just saying we don't need the package then fine, but we're not, and composer acts on the information we give it. A good example of usage would be what Laminas did, so the laminas/mvc package contains this in the json:

`    "replace": {
        "zendframework/zend-mvc": "self.version"
    }`

Reply:
The error is below:

```
Using version ^1.0 for dvsa/php-govuk-account
./composer.json has been updated
Running composer update dvsa/php-govuk-account
Loading composer repositories with package information                                                                                                            In Laminas\DependencyPlugin\DependencyRewriterV2::onPrePoolCreate
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires dvsa/php-govuk-account ^1.0 -> satisfiable by dvsa/php-govuk-account[1.0.0].
    - dvsa/php-govuk-account 1.0.0 requires paragonie/random_compat ^2 -> found paragonie/random_compat[v2.0.0, ..., v2.0.21] but these were not loaded, likely because it conflicts with another require.

You can also try re-running composer require with an explicit version constraint, e.g. "composer require dvsa/php-govuk-account:*" to figure out if any version is installable, or "composer require dvsa/php-govuk-account:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

Noticed that we need to not use ^1 of random_compat due to security vunerability. So had to ensure we only allowed ^2 and ^9.99 as supported by the upstream oauth2-client. Without this, our pipelines for installing the lowest possible packages would fail anyways (as ^1 is not considered secure).

Our package php-govuk-account would fail to install as dvsa/contracts masquerades as random_compat.

Reply:
@jerotire also, now we know what paragonie/random_compat actually does, we definitely don't need it in the require section...it's never going to be installed anyway whatever the version.

Reply:
Been on call with @JoshuaLicense who explained the true nature of "replace" and the logic behind paragonie/random_compat magic.